src/pages/index.js

```diff
@@ -74,115 +88,266 @@ const closeModalOnEsc = (event) => {
+const avatarSubmitButton = avatarModal.querySelector(".modal__submit-button");
+const avatarModalCloseButton = avatarModal.querySelector(
+  ".modal__close-button"
+);
+const avatarInput = avatarModal.querySelector("#edit-avatar-input-link");
+const profileAvatarImage = document.querySelector(".profile__avatar");
+
+avatarProfileButton.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+avatarModalCloseButton.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+avatarForm.addEventListener("submit", (event) => {
+  event.preventDefault();
+
+  const submitButton = avatarForm.querySelector(".modal__submit-button");
+  const originalText = submitButton.textContent;
+  submitButton.textContent = "Saving...";
+
+  const avatarUrl = avatarInput.value;
+
+  api
+    .editAvatar(avatarUrl)
+    .then((data) => {
+      profileAvatarImage.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      submitButton.textContent = originalText;
+    });
+});
+
+// Delete Modal
+const deleteModal = document.querySelector("#delete-modal");
+const deleteModalForm = document.querySelector("#delete-modal-form");
+const cancelButton = document.querySelector(".modal__cancel-button");
+const closeButton = document.querySelector("#close-delete-modal");
+
+let currentCardElement;
+let currentCardID;
+
+deleteModalForm.addEventListener("submit", (event) => {
+  event.preventDefault();
+
+  const submitButton = deleteModalForm.querySelector(".modal__submit-button");
+  const originalText = submitButton.textContent;
+  submitButton.textContent = "Deleting...";
+
+  api
+    .deleteCard(currentCardID)
+    .then(() => {
+      currentCardElement.remove();
+      closeModal(deleteModal);
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      submitButton.textContent = originalText;
+    });
 });
 
-cardModalImage.addEventListener("click", () => {
-  closeModal(cardModal);
+cancelButton.addEventListener("click", () => {
+  closeModal(deleteModal);
 });
 
-initialCards.forEach((card) => {
-  const cardElement = getCardElement(card);
-  cardList.append(cardElement);
+closeButton.addEventListener("click", () => {
+  closeModal(deleteModal);
 });
 
+// API Setup
+const api = new Api({
+  baseURL: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "85d7a969-7c91-4070-b3f0-78f45cb1eb48",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getAppInfo()
+  .then(([cards, userInfo]) => {
+    // debugger;
+    cards.forEach((card) => {
+      const cardElement = getCardElement(card);
+      cardElements.list.append(cardElement);
+    });
+
+    profileElements.nameElement.textContent = userInfo.name;
+    profileElements.jobElement.textContent = userInfo.about;
+    profileAvatarImage.src = userInfo.avatar;
+  })
+  .catch((err) => {
+    console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors